internal/gatewayapi/status/gatewayclass.go

```diff
@@ -76,8 +77,21 @@ func getSupportedFeatures(gatewaySuite suite.ConformanceOptions, skippedTests []
 	supportedFeatures.Delete(gatewaySuite.ExemptFeatures.UnsortedList()...)
 
 	for _, skippedTest := range skippedTests {
-		for _, feature := range skippedTest.Features {
-			supportedFeatures.Delete(feature)
+		switch conformance.GetTestSupportLevel(skippedTest) {
+		case conformance.Core:
+			for _, feature := range skippedTest.Features {
+				supportedFeatures.Delete(feature)
+			}
+		case conformance.Extended:
+			var extendedFeatures []features.SupportedFeature
+			for _, feature := range skippedTest.Features {
+				if conformance.GetFeatureSupportLevel(feature) == conformance.Extended {
+					extendedFeatures = append(extendedFeatures, feature)
+				}
+			}
+			for _, feature := range extendedFeatures {
```

Comment:
should this be out of the `skippedTest` for loop ?

Reply:
Hi @arkodg
I updated the function, let me know if it's ok